tests/unit/handle_test.py

```diff
@@ -139,94 +98,70 @@ async def test_agent_remote_handle_context() -> None:
     factory = LocalExchangeFactory()
     exchange_client = await factory.create_user_client()
     registration = await exchange_client.register_agent(EmptyAgent)
-    async with RemoteHandle(registration.agent_id) as handle:
-        with pytest.raises(ExchangeClientNotFoundError):
-            assert handle.exchange == exchange_client
-
-        with pytest.raises(ExchangeClientNotFoundError):
-            assert handle.client_id is not None
+    handle = RemoteHandle(registration.agent_id)
 
-        unbound_repr = repr(handle)
-        unbound_str = str(handle)
-
-        exchange_context.set(exchange_client)
+    with pytest.raises(ExchangeClientNotFoundError):
         assert handle.exchange == exchange_client
-        assert unbound_repr != repr(handle)
-        assert unbound_str != str(handle)
-
-
-@pytest.mark.asyncio
-async def test_agent_remote_handle_clone() -> None:
-    # We cannot use the fixture here because the fixture will create context
-    factory = LocalExchangeFactory()
-    exchange_client = await factory.create_user_client()
-    registration = await exchange_client.register_agent(EmptyAgent)
-    async with RemoteHandle(registration.agent_id, exchange_client) as handle:
-        cloned = handle.clone()
 
-        with pytest.raises(ExchangeClientNotFoundError):
-            assert cloned.exchange == exchange_client
+    exchange_context.set(exchange_client)
+    assert handle.exchange == exchange_client
```

Comment:
I think when checking the value of `handle.exchange` we should use `is` rather than `==`. Practically it has the same result here because we don't implement `__eq__` on `Handle`, but semantically they have different meanings.